include/lib.h

```diff
@@ -144,11 +145,42 @@ static inline uint64_t rdmsr(uint32_t msr_idx) {
     return (((uint64_t) high) << 32) | low;
 }
 
+/*
+ * read the msr_idx msr; the result is valid iff *read is true
+ */
+static inline uint64_t rdmsr_safe(uint32_t msr_idx, bool *read) {
+    volatile bool _read = false;
+    uint32_t low, high;
+
+    asm volatile("1: rdmsr; movq $1, %[read];"
```

Comment:
this code could be made simpler with `asm goto` directive, the supplied in label can point to the error condition.